### PR TITLE
feat(health): add wrapper for cordova-plugin-health

### PR DESCRIPTION
**Pull request by alx-andru**
Add support for [cordova-health-plugin](https://github.com/dariosalvi78/cordova-plugin-health)

### CODE REVIEW

File: src/plugins/health.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
+import {Plugin, Cordova} from './plugin';
+
+/**
+ * @param: startDate: {type: Date}, start date from which to get data
+ * @param: endDate: {type: Date}, end data to which to get the data
+ * @param: dataType: {type: String}, the data type to be queried (see below for supported data types)
```
Comment (ihadeed) [inline src/plugins/health.ts]: Documentation for interfaces should look like this:

```
export interface QueryOptions { 
   ...
   /**
   * The data type to be queried
   */
   dataType: string;
   ...
}
```

Each property needs to have its definition right above it, there is no need to have the type there as it's defined in the typings and the doc generator is able to read that.

File: src/plugins/health.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+ * import { Health } from 'ionic-native';
+ *
+ * ```
+ */
+@Plugin({
+  pluginName: 'Health',
+  plugin: 'cordova-plugin-health',
+  pluginRef: 'navigator.health',
+  repo: 'https://github.com/dariosalvi78/cordova-plugin-health',
+  install: ''
```
Comment (ihadeed) [inline src/plugins/health.ts]: `install: ''` should be removed

File: src/plugins/health.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+   * It is necessary that the app's package name and the signing key are registered in the Google API console.
+   * In Android, be aware that if the activity is destroyed (e.g. after a rotation) or is put in background,
+   * the connection to Google Fit may be lost without any callback. Going through the authorization will ensure that
+   * the app is connected again.
+   * In Android 6 and over, this function will also ask for some dynamic permissions if needed
+   * (e.g. in the case of "distance", it will need access to ACCESS_FINE_LOCATION).
+   *
+   * @param {Array<String>} datatypes a list of data types you want to be granted access to
+   *
+   * @return successCallback: {type: function}, called if all OK
```
Comment (ihadeed) [inline src/plugins/health.ts]: This should be:
```
@return {Promise<any>}
```

File: src/plugins/health.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+   * errorCallback: {type: function(err)}, called if something went wrong,
+   * err contains a textual description of the problem
+
```
Comment (ihadeed) [inline src/plugins/health.ts]: Extra lines like this one should be removed

File: src/plugins/health.ts [hunk 5]
```diff
@@ -1,60 +1,156 @@
 import {Plugin, Cordova} from './plugin';
 
-/**
- * @param: startDate: {type: Date}, start date from which to get data
- * @param: endDate: {type: Date}, end data to which to get the data
- * @param: dataType: {type: String}, the data type to be queried (see below for supported data types)
- */
+/*
```
Comment (ihadeed) [inline src/plugins/health.ts]: This large block of documentation is not necessary. It won't be visible on the site anyways.

We can add a link in the main documentation block telling people to visit the plugin's repo to learn more about the valid data types. Or this table can be under `@advanced` tag. Example:

```
/**
* @name PluginName
* @description ...
* @usage ...
* @advanced
* Put the table here 
*/
```

File: src/plugins/health.ts [hunk 6]
```diff
@@ -201,7 +201,7 @@ export class Health {
    *
    * This function is only available on Android.
    *
-   * @return successCallback: {type: function()}, called if the function was called
+   * @return  {Promise<any>} {type: function()}, called if the function was called
```
Comment (ihadeed) [inline src/plugins/health.ts]: All of this should be removed: 
```
 {type: function()}, called if the function was called
     * errorCallback: {type: function(err)}, called if something went wrong		     * errorCallback: {type: function(err)}, called if something went wrong
```

All what it needs is `@return {Promise<any>}` 

You can add additional description such as: 
```
@returns {Promise<any>} returns a promise that resolves when (something happens) and rejects if (something else happens).
```

**Comment by alx-andru** [comment]
@ihadeed thx for your review comments. I modified my fork and added the changes. I think I used the review process of github in a wrong way, please let me know if there are further changes or adjustments to make.

**Pull request merged**